### PR TITLE
Final issues fixed (main)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: src/handler.cpp [hunk 1]
```diff
@@ -322,22 +321,35 @@ namespace
                         fs::server::remove_metadata(conn, path, {*attribute_name,  std::to_string(value)});
                     }
                 }
-            });
+            };
+
+            if (is_group(*rei.rsComm, *username)) {
+                irods::experimental::scoped_permission sp{*rei.rsComm, path, fs::perms::write};
+                the_work();
```
Comment (korydraughn) [inline src/handler.cpp:328]: use a better name

File: src/handler.cpp [hunk 2]
```diff
@@ -322,22 +321,35 @@ namespace
                         fs::server::remove_metadata(conn, path, {*attribute_name,  std::to_string(value)});
                     }
                 }
-            });
+            };
+
+            if (is_group(*rei.rsComm, *username)) {
```
Comment (korydraughn) [inline src/handler.cpp:326]: this should be part of the user administration library

File: src/handler.cpp [hunk 3]
```diff
@@ -423,6 +426,17 @@ namespace
             throw std::out_of_range{error_msg};
         }
     }
+
+    auto is_group(rsComm_t& _conn, const std::string_view _entity_name) -> bool
+    {
+        const auto gql = fmt::format("select USER_TYPE where USER_NAME = '{}'", _entity_name);
+
+        for (auto&& row : irods::query{&_conn, gql}) {
+            return "rodsgroup" == row[0]; // Can we assume that this query will always return zero or one row?
```
Comment (korydraughn) [inline src/handler.cpp:435]: see question in trailing comment

Comment (trel) [reply src/handler.cpp:435]: yes, a user can only be of one type at a time.

File: src/handler.cpp (hunk 2 above)
Comment (korydraughn) [reply src/handler.cpp:326]: this is already provided by the library in the form of `exists(group)`.

the implementations should be compared to see if one is better than the other.

**Comment by korydraughn** [comment]
any issues with this PR?

File: packaging/test_rule_engine_plugin_logical_quotas.py [hunk 4]
```diff
@@ -663,6 +663,46 @@ def test_recalculating_totals_produce_the_correct_results_in_a_multi_replica_sce
...
+                    self.admin1.assert_icommand(['imeta', 'ls', '-C', group_home_col], 'STDOUT', [
+                        'attribute: irods::logical_quotas::total_size_in_bytes',
+                        'attribute: irods::logical_quotas::total_number_of_data_objects'
+                    ])
+
+                    json_string = json.dumps({'operation': 'logical_quotas_stop_monitoring_collection', 'collection': group_home_col})
+                    self.admin2.assert_icommand(['irule', '-r', 'irods_rule_engine_plugin-logical_quotas-instance', json_string, 'null', 'null'])
+
+                finally:
+                    self.admin1.run_icommand(['iadmin', 'rmgroup', group_name])
```
Comment (trel) [inline packaging/test_rule_engine_plugin_logical_quotas.py:704]: does removing the group remove the group's home collection?    or remove those AVUs?

**Review by trel — COMMENTED** [review]
looks good to me.

just a note... so much duplicate/identical code... is that something for a future refactor/abstraction?

File: packaging/test_rule_engine_plugin_logical_quotas.py (hunk 4 above)
Comment (korydraughn) [reply packaging/test_rule_engine_plugin_logical_quotas.py:704]: I'm not sure if the home collection is removed. The AVUs are removed by the call to stop monitoring the collection just above.

**Comment by korydraughn** [comment]
I'm guessing you're referring to the exception cases.

I hadn't planned on adjusting those. Replacing the function calls made it easier to control what happens in the exception case.

**Comment by korydraughn** [comment]
The only code I see that could be a little better is code dealing with `is_group` and `switch_user`.

File: packaging/test_rule_engine_plugin_logical_quotas.py (hunk 4 above)
Comment (trel) [reply packaging/test_rule_engine_plugin_logical_quotas.py:704]: if rerunning the same test more than once works - i'm good.

**Comment by trel** [comment]
yeah, just that those blocks are all relatively identical.    future work, if ever.

# em if you're happy.

File: packaging/test_rule_engine_plugin_logical_quotas.py (hunk 4 above)
Comment (korydraughn) [reply packaging/test_rule_engine_plugin_logical_quotas.py:704]: yes. the test handles being rerun multiple times.

**Comment by korydraughn** [comment]
i've consolidated the duplicate exception blocks into different functions.

merging now.

**Pull request merged**